lib/adbc_column.ex

```diff
@@ -509,6 +515,40 @@ defmodule Adbc.Column do
     preprocess_decimal(bitwidth, precision, scale, rest, [nil | acc])
   end
 
+  defp preprocess_decimal(bitwidth, precision, scale, [integer | rest], acc)
+       when is_integer(integer) do
+    if Decimal.coef_length(integer) > precision do
+      raise Adbc.Error,
+            "`#{Integer.to_string(integer)}` cannot be fitted into a decimal#{Integer.to_string(bitwidth)} with the specified precision #{Integer.to_string(precision)}"
+    else
+      coef = trunc(integer * :math.pow(10, scale))
+      acc = [<<coef::signed-integer-little-size(bitwidth)>> | acc]
+      preprocess_decimal(bitwidth, precision, scale, rest, acc)
+    end
+  end
+
+  defp preprocess_decimal(bitwidth, precision, scale, [float | rest], acc) when is_float(float) do
```

Comment:
I don't think we should support floats because they are ultimately imprecise. I would support only integers.